### PR TITLE
Filter on events table

### PR DESCRIPTION
**Pull request by titus6012000**
Add filter for the log level and the contents of the log himself.

### CODE REVIEW

File: src/components/events-table.js [hunk 1]
```diff
@@ -59,25 +64,145 @@ function inputDataRow(processEvent) {
...
+                                            type="text"
+                                            defaultValue={levelFilter}
+                                            InputLabelProps={{
+                                                shrink: true,
+                                            }}
+                                            onChange={handleLevelChange}
+                                        />
+                                    </MenuItem>
+                                </Menu>
+                            </span>
```
Comment (TheMaskedTurtle) [inline src/components/events-table.js:157]: I think you can extract a component for that to make it common for level and message

File: src/components/events-table.js [hunk 2]
```diff
@@ -59,25 +64,145 @@ function inputDataRow(processEvent) {
...
+            event.currentTarget.value.toUpperCase()
+        );
+    };
+
+    const filterProcessEvent = (currentEventFIlter, currentLogFilter) => {
+        let filtered;
+        filtered = processEvents.filter(
+            (event) =>
+                event.level.includes(currentEventFIlter) &&
+                event.message.toUpperCase().includes(currentLogFilter)
```
Comment (TheMaskedTurtle) [inline src/components/events-table.js:102]: It is a bit confusing what is in uppercase ad what is not. I would rather call the uppercase only in the comparison method, so you're sure you're comparing the good elements. This way you can store the value without uppercase for logFilter and levelFilter and you can pass these values directly to the method

File: src/components/events-table.js [hunk 3]
```diff
@@ -59,25 +64,145 @@ function inputDataRow(processEvent) {
...
+        setLevelFilter(event.currentTarget.value.toUpperCase());
+        filterProcessEvent(event.currentTarget.value.toUpperCase(), logFilter);
+    };
+
+    const handleLogMenuClick = (event) => {
+        setAnchorLogEl(event.currentTarget);
+    };
+    const handleLogChange = (event) => {
+        setLogFilter(event.currentTarget.value.toUpperCase());
+        filterProcessEvent(
```
Comment (TheMaskedTurtle) [inline src/components/events-table.js:91]: I would rather call the filter method in the rendering directly, so we don't have to define an additional state filteredProcessEvents, and we're sure we will not forget to add the filtering in the future.

File: src/components/events-table.js [hunk 4]
```diff
@@ -59,25 +64,145 @@ function inputDataRow(processEvent) {
...
+                                    anchorEl={anchorLogEl}
+                                    keepMounted
+                                    open={Boolean(anchorLogEl)}
+                                    onClose={handleCloseLog}
+                                >
+                                    <MenuItem>
+                                        <TextField
+                                            id="levelTextField"
+                                            label={
+                                                <FormattedMessage id="filterOnLog" />
```
Comment (TheMaskedTurtle) [inline src/components/events-table.js:189]: Same, you have to define this label

**Review by TheMaskedTurtle — CHANGES_REQUESTED** [review]

File: src/components/events-table.js [hunk 5]
```diff
@@ -59,25 +64,145 @@ function inputDataRow(processEvent) {
...
+                                    anchorEl={anchorLevelEl}
+                                    keepMounted
+                                    open={Boolean(anchorLevelEl)}
+                                    onClose={handleCloseLevel}
+                                >
+                                    <MenuItem>
+                                        <TextField
+                                            id="levelTextField"
+                                            label={
+                                                <FormattedMessage id="filterOnLevel" />
```
[the remark this answers is not in the document]
Comment (titus6012000) [reply src/components/events-table.js:146]: that's right. I did not split correctly my work in the multiple branch. The label is defined in the PR create a global view

File: src/components/events-table.js (hunk 4 above)
Comment (titus6012000) [reply src/components/events-table.js:189]: that's right. I did not split correctly my work in the multiple branch. The label is defined in the PR create a global view

File: src/components/events-table.js [hunk 6]
```diff
@@ -59,25 +64,145 @@ function inputDataRow(processEvent) {
 }
 
 const EventsTable = ({ taskData }) => {
-    let processEvents = taskData === null ? [] : taskData.processEvents;
+    const [anchorLevelEl, setAnchorLevelEl] = React.useState(null);
+    const [anchorLogEl, setAnchorLogEl] = React.useState(null);
+
+    const [processEvents] = React.useState(taskData.processEvents);
+    const [filtredProcessEvents, setFiltredProcessEvents] = React.useState(
```
[the remark this answers is not in the document]
Comment (titus6012000) [reply src/components/events-table.js:71]: the typo has been corrected

File: src/components/events-table.js [hunk 7]
```diff
@@ -59,25 +64,145 @@ function inputDataRow(processEvent) {
...
+    };
+    const handleLogChange = (event) => {
+        setLogFilter(event.currentTarget.value.toUpperCase());
+        filterProcessEvent(
+            levelFilter,
+            event.currentTarget.value.toUpperCase()
+        );
+    };
+
+    const filterProcessEvent = (currentEventFIlter, currentLogFilter) => {
```
[the remark this answers is not in the document]
Comment (titus6012000) [reply src/components/events-table.js:97]: the typo has been corrected

File: src/components/events-table.js (hunk 2 above)
Comment (titus6012000) [reply src/components/events-table.js:102]: In fact everything is in uppercase. I will modify that point in order to clearly show what is compared

File: src/components/events-table.js (hunk 1 above)
Comment (titus6012000) [reply src/components/events-table.js:157]: I extracted the filter menu and create a new component. I also use the new component in global-view-core

**Pull request closed without merging**